### PR TITLE
Modal view wallets selector

### PR DESCRIPTION
**Pull request by bogos**
- Added the useink library
- Replaced how users are listed using useink
- Replaced chain description using useink
- Replaced api provider using useink
- Updated the interface according to figma
- It is now possible to select a list of wallets,
  - installed
  - to be installed
- Addresses are listed according to the wallet selected
- Added a button to disconnect the wallet

### CODE REVIEW

File: src/context/NetworkAccountsContext.tsx [hunk 1]
```diff
@@ -1,160 +1,104 @@
...
+    }))
+  }
 
-    return () => {
-      document.removeEventListener(
-        WalletConnectionEvents.walletConnectInit,
-        () => loadAccounts(state, setState)
-      )
+  const setCurrentAccount = (accountAddress: string) => {
+    if (accountAddress === 'disconnect') {
```
Comment (henrypalacios) [inline src/context/NetworkAccountsContext.tsx:62]: Let's move this `disconnect` string to an `OPTION_FOR_DISCONNECTING` constant at the beginning of this file.

File: src/context/NetworkAccountsContext.tsx [hunk 2]
```diff
@@ -1,160 +1,104 @@
...
+      disconnectWallet()
+      return
     }
-  }, [state])
+    const account: WalletAccount = accounts?.filter(
```
Comment (henrypalacios) [inline src/context/NetworkAccountsContext.tsx:66]: I think this is unnecessary, you could just use the useWallet hook to get the current `WalletAccount`.
```typescript
const { account: currentAccount } = useWallet();
```

File: src/context/NetworkAccountsContext.tsx [hunk 3]
```diff
@@ -1,160 +1,104 @@
...
+  }
 
-  function setCurrentAccount(account: string) {
-    setState(prev => ({ ...prev, currentAccount: account }))
-    document.dispatchEvent(
-      new CustomEvent(WalletConnectionEvents.changeAccountAddress)
-    )
+  const setCurrentWallet = async (wallet: Wallet) => {
+    const { extensionName, getAccounts } = wallet
+    await wallet.enable('contract-wizard')
```
Comment (henrypalacios) [inline src/context/NetworkAccountsContext.tsx:75]: Use the app name that is described in the constant configuration [`DAPP_CONFIG`](https://github.com/protofire/polkadot-contract-wizard/blob/feature_add_account_selector/src/constants/config.ts#L12)

File: src/domain/DomainEvents.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 export const enum WalletConnectionEvents {
   walletConnectInit = 'walletConnectInit',
-  changeAccountAddress = 'changeAccountAddress'
+  changeAccountAddress = 'changeAccountAddress',
+  listAllWallets = 'listAllWallets'
```
Comment (henrypalacios) [inline src/domain/DomainEvents.ts:4]: The idea of the Domain events is that another part of the application can fire something when listening to them, what will fire this new `listAllWallets` event? if it is not necessary to remove it

File: src/types/wallet.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+export type SubscriptionFn = (
```
Comment (henrypalacios) [inline src/types/wallet.ts:1]: This tiping should also be part of `/infraestructura/useink/walletTypes.ts` as it is a representation of a third party interfaces.

File: src/view/components/ModalWallet/index.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  Modal,
+  Typography
+} from '@mui/material'
+import { Wallet } from '@/types'
+
+const modalStyle = {
```
Comment (henrypalacios) [inline src/view/components/ModalWallet/index.tsx:16]: 👍

**Review by henrypalacios — CHANGES_REQUESTED** [review]
Very good contribution! I leave you some changes to manage

File: src/constants/chain.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import {
+  Astar,
+  Chain,
+  RococoContractsTestnet,
+  ShibuyaTestnet,
+  ShidenKusama
+} from './chains'
```
[the remark this answers is not in the document]
Comment (bogos) [reply src/constants/chain.ts:7]: Done

File: src/view/components/WalletConnectButton/AccountsSelect.tsx [hunk 8]
```diff
@@ -57,38 +59,95 @@ const StyledMenuItem = styled(MenuItem)<MenuItemProps>(() => ({
...
+  if (!accounts)
+    return (
+      <StyledSelect
+        value={'Select Account...'}
+        placeholder="Select Account..."
+      ></StyledSelect>
+    )
+  const allAccounts = [
+    ...accounts,
+    { name: 'disconnect', address: 'disconnect' }
```
[the remark this answers is not in the document]
Comment (bogos) [reply src/view/components/WalletConnectButton/AccountsSelect.tsx:84]: Done

File: src/types/wallet.ts (hunk 5 above)
Comment (bogos) [reply src/types/wallet.ts:1]: Done

File: src/view/components/ModalWallet/index.tsx [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+              marginTop: '2rem',
+              marginBottom: '1.5rem'
+            }}
+            variant="h6"
+          >
+            Installed Wallets
+          </Typography>
+          <List disablePadding sx={listSx}>
+            {wallets
+              .filter(wallet => wallet.installed)
```
[the remark this answers is not in the document]
Comment (bogos) [reply src/view/components/ModalWallet/index.tsx:86]: Done

File: src/context/NetworkAccountsContext.tsx [hunk 10]
```diff
@@ -1,160 +1,104 @@
...
   currentAccount?: string
   jsonRpc: typeof jsonrpc
-  apiStatus: NetworkState
   api?: ApiPromise
   apiError?: string
-  accountStatus: NetworkState // keyring state
-  keyring?: Keyring
   chainInfo?: ChainProperties
+  accountStatus: NetworkState
+  walletLogo: WalletLogoProps
```
[the remark this answers is not in the document]
Comment (bogos) [reply src/context/NetworkAccountsContext.tsx:21]: Done

**Pull request merged**